### PR TITLE
Migrate small utility modules to ESM

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

Ref #5291

> Is there anything in the PR that needs further explanation?

This change rewrites small utility modules (less dependent on 3rd-party packages) to ESM.

### CODE REVIEW

File: lib/utils/getSchemeFromUrl.cjs [hunk 1]
```diff
@@ -1,15 +1,14 @@
 'use strict';
 
-const { URL } = require('url');
-
```
Comment (ybiquitous) [inline lib/utils/getSchemeFromUrl.cjs:4]: [note] `URL` is now a global object. See https://nodejs.org/api/url.html#class-url

File: lib/utils/isAfterSingleLineComment.mjs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import isSharedLineComment from './isSharedLineComment.mjs';
+
+/**
+ * @param {import('postcss').Node} node
+ * @returns {boolean}
```
Comment (ybiquitous) [inline lib/utils/isAfterSingleLineComment.mjs:5]: [note] Specifying a return type makes code clearer as documentation. Also, it's valid for type-check.

File: lib/__tests__/ignore.test.mjs [hunk 3]
```diff
@@ -176,7 +176,7 @@ test('specified `ignorePath` file ignoring one file', async () => {
 			},
 			ignorePath: [fixtures('ignore.txt')],
 		}),
-	).rejects.toThrow(AllFilesIgnoredError); // all files ignore
+	).rejects.toThrow(new AllFilesIgnoredError());
```
Comment (ybiquitous) [inline lib/__tests__/ignore.test.mjs:179]: [note] Just specifying a constructor fails the test. `new` is needed.

**Review by jeddy3 — APPROVED** [review]
Excellent, thank you!

**Pull request merged**